Tests/Tests.iOS/PreserveFromStripping.xml

```diff
@@ -0,0 +1,6 @@
+<linker>

```

Comment:
We should not have this. If the managed linker strips those, we should annotate them with `[Preserve(AllMembers = true)]`.

Reply:
@fealebenpae suggested that it's better to avoid adding that attribute when in the main assembly unless strictly necessary. In our case I get the setter of all callbacks for `AutomaticRecoveryHandler` and `DiscardLocalResetHandler` stripped. I just supposed that users should not run in the same issue. I just spotted this stripping problem because I used reflections in tests, it would otherwise have gone unnoticed.
So I'm not sure what the best course of action is here. When dealing with user facing API, should we preserve more often then not or keep it at a minimum?

Reply:
Hm... good point. I didn't realize it's due to reflection that those were stripped. In that case, let's preserve them by using them in the tests rather than with a xml file. Those can easily go out of sync with the code they're preserving and are a bit annoying to maintain (e.g. you need to also add this for MAUI and Unity). I think we can preserve them like:

```csharp
[Preserve]
static SessionTests()
{
    var preserveHandler = new AutomaticRecoveryHandler
    {
        OnBeforeReset = (...) => { },
	...
    }
}
```